packages/docs-app/src/examples/select-examples/multiSelectExample.tsx

```diff
@@ -173,4 +210,16 @@ export class MultiSelectExample extends React.PureComponent<IExampleProps, IMult
     }
 
     private handleClear = () => this.setState({ films: [] });
+
+    private arrayContainsFilm(films: IFilm[], filmToFind: IFilm): boolean {
+        return films.find(f => f.title === filmToFind.title) !== undefined;
+    }
+
+    private addFilmToArray(films: IFilm[], filmToAdd: IFilm) {
```

Comment:
This reconciliation logic now falls on the users—is it possible to keep this inside the multiselect component? i.e. wrap the passed-in `onItemSelect` and move the logic for maintaining the union of {passed-in items, newly-created items}

Reply:
@shuyangli It's a good thought, but it seems like the component should remain as data-agnostic as possible. Tracking created items would violate that. @adidahiya what do you think?

Reply:
In particular, a major blocker for doing that is that if a user passes in a totally different set of `items`, wiping away the ones from before, then our components won't have any way to know if they should clear their internally tracked list of `createdItems` (which would be stored in `this.state`, presumably). The mutually dependent `props.items` and `state.createdItems` could easily get out of sync and cause more problems than they solve.

Reply:
yeah I agree with @cmslewis here, I would prefer to do less data handling in the component and keep it more presentation-focused. I know it's pretty verbose for consumers this way, but we can always add additional conveniences on top of this approach later on after we see concrete use cases of this feature